api/types.proto

```diff
@@ -30,12 +30,32 @@ message Annotations {
 	repeated IndexEntry indices = 4 [(gogoproto.nullable) = false];
 }
 
+message GenericString {
+	string kind = 1;
+	string value = 2;
+}
+
+message GenericDiscrete {
+	string kind = 1;
+	int64 value = 2;
```

Comment:
That changed because of reconciliation where you can go negative and this gets corrected later in the constraint manager